devlib/trace/ftrace.py

```diff
@@ -177,9 +187,16 @@ def start(self):
         if self._reset_needed:
             self.reset()
 
+        if self.tracer is not None and 'function' in self.tracer:
+            function_string = self.function_string
+        else:
+            function_string = ''
+
```

Comment:
If we're changing to use a local version of `function_string` here would it make sense for line 215 to also be updated for consistency?

Reply:
Using `function_string` to write to sysfs would lead to write an empty string when `self.tracer is None`, which is true when executing line 215. That local `function_string` is only meant to be given to trace-cmd. Maybe I should rename it to `tracecmd_functions` instead.